### PR TITLE
Add amount claim

### LINKED ISSUE #212: Add OutputsToClaim::Amount variant
## Description

Add Amount variant to OutputsToClaim

## Motivation

Claim only outputs where we get more amount, for example basic output with expiration unlock condition

## Requirements

1. Add `Amount` variant to https://github.com/iotaledger/iota-sdk/blob/b5a72f8aea9be4dc4bc0b59087e910f6d24bcc35/sdk/src/wallet/account/operations/output_claiming.rs#L25
2. Handle this case here https://github.com/iotaledger/iota-sdk/blob/b5a72f8aea9be4dc4bc0b59087e910f6d24bcc35/sdk/src/wallet/account/operations/output_claiming.rs#L73-L87 and only add outputs where we get more amount

## Are you planning to do it yourself in a pull request?

No.

### PR DESCRIPTION
**Pull request by DaughterOfMars**
# Description of change

Adds the Amount claiming variant to claim outputs that net amount to the account.

## Links to any relevant issues

- Fixes #212 

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How the change has been tested

Added a test case.

### CODE REVIEW

File: sdk/src/wallet/account/operations/output_claiming.rs [hunk 1]
```diff
@@ -27,7 +27,8 @@ pub enum OutputsToClaim {
     MicroTransactions = 1,
     NativeTokens = 2,
     Nfts = 3,
-    All = 4,
+    Amount = 4,
+    All = 5,
```
Comment (thibault-martinez) [inline sdk/src/wallet/account/operations/output_claiming.rs]: Do they actually have to have discriminants ?

File: sdk/src/wallet/account/operations/output_claiming.rs [hunk 2]
```diff
@@ -50,88 +51,55 @@ impl Account {
...
+                            &[],
+                            output_data,
+                            local_time,
+                            // Not relevant without alias addresses
+                            None,
+                        )?
+                    {
+                        match outputs_to_claim {
+                            OutputsToClaim::MicroTransactions => {
+                                if let Some(sdr) = unlock_conditions.storage_deposit_return() {
```
Comment (Thoralf-M) [inline sdk/src/wallet/account/operations/output_claiming.rs:77]: I know, not the part you added, but what do you think about adding this here?
```suggestion

                                if let Some(sdr) = unlock_conditions.storage_deposit_return() {
                                    // If expired, it's not a micro transaction anymore
                                    if unlock_conditions.is_expired(local_time) {
                                        continue;
                                    }
```

File: sdk/src/wallet/account/operations/output_claiming.rs (hunk 1 above)
Comment (DaughterOfMars) [reply sdk/src/wallet/account/operations/output_claiming.rs]: I don't think so. I was wondering the same thing tbh

**Review by Thoralf-M — DISMISSED** [review]

File: sdk/src/wallet/account/operations/output_claiming.rs [hunk 3]
```diff
@@ -50,88 +50,63 @@ impl Account {
         let mut output_ids_to_claim: HashSet<OutputId> = HashSet::new();
         for (output_id, output_data) in &account_details.unspent_outputs {
             // Don't use outputs that are locked for other transactions
-            if !account_details.locked_outputs.contains(output_id) {
-                if let Some(output) = account_details.outputs.get(output_id) {
-                    match &output.output {
-                        Output::Basic(basic_output) => {
```
Comment (thibault-martinez) [inline sdk/src/wallet/account/operations/output_claiming.rs:56]: Is it correct to remove Basic/Nft match? Isn't it going to pick aliases and foundries?

**Review by Thoralf-M — APPROVED** [review]

File: sdk/src/wallet/account/operations/output_claiming.rs (hunk 3 above)
Comment (Thoralf-M) [reply sdk/src/wallet/account/operations/output_claiming.rs:56]: Good catch, still needs to be handled

**Review by Thoralf-M — CHANGES_REQUESTED** [review]
Unresolved comment

File: sdk/src/wallet/account/operations/output_claiming.rs (hunk 3 above)
Comment (thibault-martinez) [reply sdk/src/wallet/account/operations/output_claiming.rs:56]: Maybe you can simply filter the iterator by `is_basic || is_nft` @Alexandcoats

**Review by Thoralf-M — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**